### PR TITLE
Add GitHub Actions CI

### PR DESCRIPTION
**Pull request by NickGoog**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        set -x
+        pip install .
+        git submodule update --init --recursive
+    - name: Lint
+      if: ${{ matrix.python-version  == 3.6 }}
+      run: |
+        set -x
+        pip install pylint
+        pip install yapf
+        test/run_pylint.sh;
```
Comment (dilipped) [inline .github/workflows/ci.yml:34]: Why do these require semi-colon in the end, but other commands don't?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        set -x
+        pip install .
+        git submodule update --init --recursive
```
Comment (dilipped) [inline .github/workflows/ci.yml:27]: Pip should install all the dependencies. Do we still need git submodule update?

Comment (NickGoog) [reply .github/workflows/ci.yml]: Error without:
```
+ python gsutil version -l
Traceback (most recent call last):
  File "gsutil", line 21, in <module>
    gsutil.RunMain()
  File "/home/runner/work/gsutil/gsutil/gsutil.py", line 121, in RunMain
    import gslib.__main__
  File "/home/runner/work/gsutil/gsutil/gslib/__main__.py", line 94, in <module>
    from gslib.command_runner import CommandRunner
  File "/home/runner/work/gsutil/gsutil/gslib/command_runner.py", line 64, in <module>
    from gslib.tests.util import HAS_NON_DEFAULT_GS_HOST
  File "/home/runner/work/gsutil/gsutil/gslib/tests/util.py", line 38, in <module>
    import mock_storage_service  # From boto/tests/integration/s3
ModuleNotFoundError: No module named 'mock_storage_service'
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (NickGoog) [reply .github/workflows/ci.yml]: Good catch

**Review by dilipped — APPROVED** [review]

**Pull request merged**